lte/cloud/go/services/subscriberdb/servicers/subscriberdb_servicer.go

```diff
@@ -52,8 +57,8 @@ func (s *subscriberdbServicer) ListSubscribers(ctx context.Context, req *lte_pro
 	if !gateway.Registered() {
 		return nil, status.Errorf(codes.PermissionDenied, "gateway is not registered")
 	}
-	networkID := gateway.NetworkId
-	gatewayID := gateway.LogicalId
+	networkID := gateway.GetNetworkId()
+	gatewayID := gateway.GetLogicalId()
```

Comment:
Why the change? Generally don't use "guarded" getters unless you expect the object (gateway) to possibly be nil

Reply:
This practice of guarding against the gateway being `nil` has been recommended by @emakeev in a [previous review](https://github.com/magma/magma/pull/7302#discussion_r645828623); wondering what is the downside to using a guarded function like this?

Reply:
Gateway won't ever be nil here -- and if it is, it's a bug and we'll fix it

IMO it's about communicating to future readers your assumptions about `gateway` -- when you use guarded getters, you communicate that `gateway` is expected to possibly be nil. But it's not, we expect it to always be non-nil here

Unfortunately Go doesn't handle nil-ness very well -- languages like Rust build nil-ness into the static type system and it makes this type of thing much easier. That said, Go's method is more concise, so let's stick with the conciseness for now -- we don't expect `gateway` to be nil ever, so communicate that properly by not using the guarded getter

Reply:
To follow up -- if `gateway` is nil (somehow), then the rest of the function runs with empty string for networkID and gatewayID, which is useless and potentially dangerous. So realistically the solution would be just checking whether `gateway == nil`. But, we can skip that cause we "know" it's not, based on the impl of the fn that creates it

Reply:
Late to the party but also wanted to point out - I think the GetMessage() calls return a "default constructed" object if nil? Is that right? And I personally would far prefer nil and the knowledge that a field wasn't set - to a default-constructed object that is perhaps misleading or erroneous and shouldn't be operated on.